pkg/services/alerting/rule.go

```diff
@@ -140,7 +140,13 @@ func NewRuleFromDBAlert(ruleDef *models.Alert) (*Rule, error) {
 		if id, err := jsonModel.Get("id").Int64(); err == nil {
 			uid, err := translateNotificationIDToUID(id, ruleDef.OrgId)
 			if err != nil {
-				logger.Error("Unable to translate notification id to uid", "error", err.Error(), "dashboardId", model.DashboardID, "alertId", model.ID, "panelId", model.PanelID, "notificationId", id)
+				if !errors.Is(err, models.ErrAlertNotificationFailedTranslateUniqueID) {
+					return nil, ValidationError{Reason: "Failed to tranlate notification id to uid:" + err.Error(), DashboardID: model.DashboardID, AlertID: model.ID, PanelID: model.PanelID}
+				}
+
+				if logTranslationFailures {
+					logger.Warn("Unable to translate notification id to uid", "error", err.Error(), "dashboardId", model.DashboardID, "alertId", model.Name, "panelId", model.PanelID, "notificationId", id)
```

Comment:
We already know here that the error is `models.ErrAlertNotificationFailedTranslateUniqueID` so don't need to include that generic error message. Why did you change `model.ID` to `model.Name`? In case the rule is not saved to the database and have an id of zero?  

```suggestion
					logger.Warn("Unable to translate notification id to uid", "dashboardId", model.DashboardID, "alertId", model.ID, "panelId", model.PanelID, "notificationId", id)
```

Reply:
No, I've changed it because I think that a user can remember easier the name instead of the id.
But I'll keep your suggestion with the ID as it used to be before.

Reply:
Another reason for logging `model.Name` instead of `model.ID` is that if `NewRuleFromDBAlert` is called for an alert returned from a database query contains the ID:
https://github.com/grafana/grafana/blob/0b9eae7f84152e4187ced34f5c62886c7d9cf012/pkg/services/alerting/reader.go#L39

but when it is called with for an alert extracted from a dashboard JSON does not:
https://github.com/grafana/grafana/blob/0b9eae7f84152e4187ced34f5c62886c7d9cf012/pkg/services/alerting/extractor.go#L191
because the id is not stored in the JSON settings, so this is empty:
https://github.com/grafana/grafana/blob/0b9eae7f84152e4187ced34f5c62886c7d9cf012/pkg/services/alerting/extractor.go#L131